hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileCreateRequest.java

```diff
@@ -0,0 +1,348 @@
+          boolean canBeCreated = checkKeysUnderPath(omMetadataManager,
+              volumeName, bucketName, keyName);
+          if (!canBeCreated) {
+            throw new OMException("Can not create file: " + keyName + "as one" +
+                " of parent directory is not created",
+                OMException.ResultCodes.NOT_A_FILE);
+          }
+        }
+      }
+
+      // do open key
+      encryptionInfo = getFileEncryptionInfo(ozoneManager, bucketInfo);
+      omKeyInfo = prepareKeyInfo(omMetadataManager, keyArgs,
+          omMetadataManager.getOzoneKey(volumeName, bucketName,
+              keyName), keyArgs.getDataSize(), locations, encryptionInfo);
+
+    } catch (IOException ex) {
+      exception = ex;
+    } finally {
+      if (acquiredLock) {
+        omMetadataManager.getLock().releaseLock(BUCKET_LOCK, volumeName,
+            bucketName);
+      }
+    }
+
+    return prepareCreateKeyResponse(keyArgs, omKeyInfo, locations,
+        encryptionInfo, exception, createFileRequest.getClientID(),
+        transactionLogIndex, volumeName, bucketName, keyName, ozoneManager,
+        OMAction.CREATE_FILE);
+  }
+
+
+
+  /**
+   * Check if any keys exist under given path.
+   * @param omMetadataManager
+   * @param volumeName
+   * @param bucketName
+   * @param keyName
+   * @return if exists true, else false. If key name is one level path return
+   * true.
+   * @throws IOException
+   */
+  private boolean checkKeysUnderPath(OMMetadataManager omMetadataManager,
+      @Nonnull String volumeName, @Nonnull String bucketName,
+      @Nonnull String keyName) throws IOException {
+
+    Path parentPath =  Paths.get(keyName).getParent();
+
+    if (parentPath != null) {
+      String dbKeyPath = omMetadataManager.getOzoneDirKey(volumeName,
+          bucketName, parentPath.toString());
+
+      // First check in key table cache.
+      Iterator< Map.Entry<CacheKey<String>, CacheValue<OmKeyInfo>>> iterator =
+          omMetadataManager.getKeyTable().cacheIterator();
+
+      while (iterator.hasNext()) {
+        Map.Entry< CacheKey< String >, CacheValue< OmKeyInfo > > entry =
+            iterator.next();
+        String key = entry.getKey().getCacheKey();
+        OmKeyInfo omKeyInfo = entry.getValue().getCacheValue();
+        // Making sure that entry is not for delete key request.
+        if (key.startsWith(dbKeyPath) && omKeyInfo != null) {
+          return true;
+        }
+      }
+      try (TableIterator< String, ? extends Table.KeyValue<String, OmKeyInfo>>
+               keyIter = omMetadataManager.getKeyTable().iterator()) {
+        Table.KeyValue< String, OmKeyInfo > kv = keyIter.seek(dbKeyPath);
+
+        if (kv != null) {
+          // Check the entry in db is not marked for delete. This can happen
+          // while entry is marked for delete, but it is not flushed to DB.
+          CacheValue< OmKeyInfo > cacheValue = omMetadataManager.getKeyTable()
+              .getCacheValue(new CacheKey(kv.getKey()));
```

Comment:
The above line gives an unchecked assignment warning. The getKeyTable returns `Table<String, OmKeyInfo>`, however the type passed to `.getCacheValue` method is CacheKey. So there is a type mismatch.